simulated_data_sources.go

```diff
@@ -84,9 +81,15 @@ func (ts *TriangleSource) blockingRead() error {
 	for _, ch := range ts.output {
 		datacopy := make([]RawType, ts.cycleLen)
 		copy(datacopy, ts.onecycle)
-		seg := DataSegment{rawData: datacopy, framesPerSample: 1}
+		seg := DataSegment{
+			rawData:         datacopy,
+			framesPerSample: 1,
+			firstFramenum:   ts.lastFrameNum,
+			firstTime:       ts.lastread,
+		}
```

Comment:
This looks like a case where a `NewDataSegment` function that requires all argument would have helped.

Reply:
Oddly enough, there is such a thing. It would have helped here, but I think I avoided it because it returned a `*DataSegment` instead of a `DataSegment`. Should I be passing around the latter instead of the former? Not sure.